packages/components/button/src/button.vue

```diff
@@ -24,25 +24,34 @@
 </template>
 
 <script lang="ts">
+import isEmpty from 'lodash/isEmpty'
 import { computed, inject, defineComponent } from 'vue'
 import { useFormItem } from '@element-plus/hooks'
-import { elButtonGroupKey } from '@element-plus/tokens'
+import { elButtonGroupKey, elFormKey } from '@element-plus/tokens'
 
 import { buttonEmits, buttonProps } from './button'
 
+import type { ElFormContext } from '@element-plus/tokens'
 export default defineComponent({
   name: 'ElButton',
 
   props: buttonProps,
   emits: buttonEmits,
 
-  setup(_, { emit }) {
+  setup(props, { emit }) {
     const elBtnGroup = inject(elButtonGroupKey, undefined)
```

Comment:
How about setting default to `undefined`

Reply:
Only write el-button type="reset", no form